### PR TITLE
[v2] Update Dependencies

### PR DESCRIPTION
**Pull request by emlynmac**
# What
Update TypeScript version to 4.3.5.
Update ESLint to version 8.x.
Update other deps

# Why
Ongoing dependency updates; start of v2 is a great time to get ahead.

# How Tested
CI / Build

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/communication-react/review/stable/communication-react.api.md [hunk 1]
```diff
@@ -1470,98 +1470,98 @@ export const darkTheme: PartialTheme & CallingTheme;
...
-    SendBoxSendHovered: JSX.Element;
-    VideoTileMicOff: JSX.Element;
-    SplitButtonPrimaryActionCameraOn: JSX.Element;
-    SplitButtonPrimaryActionCameraOff: JSX.Element;
-    SplitButtonPrimaryActionMicUnmuted: JSX.Element;
-    SplitButtonPrimaryActionMicMuted: JSX.Element;
-    ContextMenuCameraIcon: JSX.Element;
-    ContextMenuMicIcon: JSX.Element;
-    ContextMenuSpeakerIcon: JSX.Element;
+    ChatMessageOptions: React_2.JSX.Element;
```
Comment (JamesBurnside) [inline packages/communication-react/review/stable/communication-react.api.md:1473]: Any idea on this change? If its removable?

Comment (JamesBurnside) [reply packages/communication-react/review/stable/communication-react.api.md:1473]: Are we somehow pulling in two versions of react types?
image

File: common/autoinstallers/rush-prettier/package.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
   "version": "1.0.0",
   "private": true,
   "dependencies": {
-    "prettier": "2.3.1",
-    "pretty-quick": "~3.1.0"
+    "prettier": "3.0.0",
+    "lint-staged": "13.2.3"
```
Comment (vhuseinova-msft) [inline common/autoinstallers/rush-prettier/package.json:7]: Is it changed because of prettier version update?

Comment (emlynmac) [reply common/autoinstallers/rush-prettier/package.json:7]: Yes, pretty-quick does not work with this version. lint-staged is also more versatile

File: packages/communication-react/review/stable/communication-react.api.md (hunk 1 above)
Comment (emlynmac) [reply packages/communication-react/review/stable/communication-react.api.md:1473]: I've seen double includes before; not sure why here.
Searching for `@types/react": "18.2` reveals only one version.

Comment (emlynmac) [reply packages/communication-react/review/stable/communication-react.api.md:1473]: Looks like the React_2 def has been there for a while.
`import { default as React_2 } from 'react';` appeared 18 months ago.

Comment (emlynmac) [reply packages/communication-react/review/stable/communication-react.api.md:1473]: There's also things like: `import { CallState as CallState_2 } from '@azure/communication-calling';`
Unclear why the doc tool chooses this.

Comment (JamesBurnside) [reply packages/communication-react/review/stable/communication-react.api.md:1473]: `CallState` is actually a conflict between a Calling SDK type and the stateful call client State type

**Review by vhuseinova-msft — APPROVED** [review]

**Review by JamesBurnside — APPROVED** [review]

**Pull request merged**